### ISSUE TITLE
Running multiple Starcounter tests on a single machine

### ISSUE THREAD

**Issue by dan31** [Starcounter/Starcounter#3972]
We are impeded in CI cluster throughput. However, the reason is not cluster itself (hence not number of machines and not their horsepower), but the fact that it is impossible to run multiple tests in parallel on a single machine (mainly because Starcounter, as it is made today, does not allow for it). By this we effectively hold vast amount of compute resource available today from being utilised.

Bad and expensive way is to throw in more VMs so that each VM runs always a single tests - no need to explain why this is not a solution.

The proper solution entails:

1. Make it possible to install and run different Starcounters in parallel / in isolation.
2. Adopt tests so that ports and other relative things are now tests parameters.
3. Buy more TC licenses to let (Yes: one TC agent can only run one job at a time; one can install multiple TC agents on a single machine - and this is what we shall do).

@alemoi please redact / enhance.

**Comment by alemoi** [comment]
> Make it possible to install and run different Starcounters in parallel / in isolation.

So far, Im thinking ONLY about "manual" installation, meaning like on build servers but not GUI installer, because of problems like VS extension etc.

> Buy more TC licenses to let (Yes: one TC agent can only run one job at a time; one can install multiple TC agents on a single machine - and this is what we shall do).

Yes. @dan31 I would also like some Web-site certificate, not self-signed, because of Java communication, do we have such?

**Comment by dan31** [comment]
@alemoi indeed, we speak about console-only / manual / zip-based installation.
Web-site certificate - I guess this is something one purchases? I am not sure whether we have purchased any certificates except those of installers. Remind Joachim on that topic next time he's around, or into chat. I guess we can simply get one.

**Comment by alemoi** [comment]
We have a code signing certificate maybe I can try to use it for teamcity.starcounter.org

**Comment by per-samuelsson** [comment]
Move this to AdminTrack?

**Comment by k-rus** [comment]
> Move this to AdminTrack?

+1

**Comment by dan31** [comment]
This one is about making it possible to run 2 Starcounters in parallel on the same machine. This is a general feature request and it belongs to the core startcounter functionality.
